packaging/docker/build-images.sh

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+reset=$(tput sgr0)
+blue=$(tput setaf 4)
+
+function logg() {
+    printf "${blue}##### $(date +"%H:%M:%S") #  %-56.55s #####${reset}\n" "${1}"
+}
+
+function pushd () {
+    command pushd "$@" > /dev/null
+}
+
+function popd () {
+    command popd > /dev/null
+}
+
+function create_fake_website_directory() {
+    fdb_binaries=( 'fdbbackup' 'fdbcli' 'fdbserver' 'fdbmonitor' )
+    fake_fdb_binaries=( 'backup_agent' 'dr_agent' 'fastrestore_tool' 'fdbdr' 'fdbrestore' )
+    logg "PREPARING WEBSITE"
+    website_directory="${script_dir}/website"
+    rm -rf "${website_directory}"
+    mkdir -p "${website_directory}/downloads/${fdb_version}/linux/bin"
+    pushd "${website_directory}/downloads/${fdb_version}/linux/bin" || exit 127
+    ############################################################################
+    # there are four intended paths here:
+    # 1) fetch the unstripped binaries and client library from artifactory_base_url
+    # 2) fetch the stripped binaries and multiple client library versions
+    #    from artifactory_base_url
+    # 3) copy the unstripped binaries and client library from the current local
+    #    build_output of foundationdb
+    # 4) copy the stripped binaries and client library from the current local
+    #    build_output of foundationdb
+    ############################################################################
+    logg "FETCHING BINARIES"
+    case "${stripped_binaries_and_from_where}" in
+        "unstripped_artifactory")
+            logg "DOWNLOADING BINARIES TAR FILE"
+            curl -Ls "${artifactory_base_url}/${fdb_version}/release/api/foundationdb-binaries-${fdb_version}-linux.tar.gz" | tar -xzf -
+            ;;
+        "stripped_artifactory")
+            for file in "${fdb_binaries[@]}"; do
+                logg "DOWNLOADING ${file}"
+                curl -Ls "${artifactory_base_url}/${fdb_version}/release/files/linux/bin/${file}" -o "${file}"
+                chmod 755 "${file}"
+            done
+            ;;
+        "unstripped_local")
+            for file in "${fdb_binaries[@]}"; do
+                logg "COPYING ${file}"
+                cp -pr "${build_output_directory}/bin/${file}" "${file}"
+                chmod 755 "${file}"
+            done
+            ;;
+        "stripped_local")
+            for file in "${fdb_binaries[@]}"; do
+                logg "COPYING ${file}"
+                cp -pr "${build_output_directory}/packaging/bin/${file}" "${file}"
+                chmod 755 "${file}"
+            done
+            ;;
+    esac
+    # dont download files that are binary duplicates of fdbbackup, recreate the
+    # symlinks (to fdbbackup in the same directory)
+    logg "CREATING fdbbackup SYMLINKS"
+    for fake in "${fake_fdb_binaries[@]}"; do
+        logg "CREATING ${fake}"
+        ln -sf fdbbackup "${fake}"
```

Comment:
What is the benefit of having them symlinked instead of having multiple copies? I guess a reduced image size? I think that should be fine and shouldn't cause any issues.

Reply:
reduced image size, especially with the binaries that contain the debug symbols. Our RPM and DEB packages maintain the symlinks. I don't see a good reason not keep the symlinks in the container images.